### ISSUE TITLE
Change default cluster domain (`--cluster-domain` flag) for upstream services to `cluster.local`

### ISSUE THREAD

**Issue by pmalek** [Kong/kubernetes-ingress-controller#6756]
### Problem statement

#6697 introduced a new flag to set the cluster domain which is used for setting the name on upstream services.

This issue tracks the change of the default value for that flag from `""` to `cluster.local`.

**Comment by sandromodarelli** [comment]
@pmalek @programmer04 wdyt about punting this to KO 2.0.0?

**Comment by programmer04** [comment]
Yup, it's reasonable. I'd even do it for KO `v2.0.0-alpha.1`, because it's just one place change, which is a breaking change. So sonner is  better

**Comment by alacuku** [comment]
Done in: https://github.com/Kong/kong-operator/pull/1870
